### ISSUE TITLE
[FEATURE] Run same tests for security enabled or disabled cluster

### ISSUE THREAD

**Issue by owaiskazi19** [opensearch-project/flow-framework#478]
### Is your feature request related to a problem?
Currently, we run different test suites for security enabled tests [here](https://github.com/opensearch-project/flow-framework/blob/main/src/test/java/org/opensearch/flowframework/rest/FlowFrameworkSecureRestApiIT.java#L27), which is mostly checking for the roles and not actually testing the APIs with the response. Whereas our [integ tests](https://github.com/opensearch-project/flow-framework/tree/main/src/test/java/org/opensearch/flowframework/rest) does that. 

### What solution would you like?
We should run the same integ tests suite for the plugin with security-enabled or disabled cluster with just changing the local cluster url to pick either `https` or `http` protocol based on the flag `security-enabled=true` or `security-enabled=false`.

### What alternatives have you considered?
The current test suite which just checks for the role based access for security enabled cluster.

**Comment by joshpalis** [comment]
I agree with your sentiment, but thinking about this again, overall I think having more tests is always good, especially as we add support for more plugins in the future. Running all of our integration tests with security enabled should at the very least help us catch issues early on that would otherwise only be caught during full distribution builds.

**Comment by minalsha** [comment]
this is a nice to have and I dont think this is a blocker for 2.13.
